core-foundation/src/dictionary.rs

```diff
@@ -82,13 +89,23 @@ impl CFDictionary {
         }
     }
 
+    /// Similar to `find` but acts on a higher level, automatically converting from any `TCFType`
+    /// to the raw pointer of its concrete TypeRef.
+    #[inline]
+    pub fn find2<X, K: TCFType<*const X>>(&self, key: &K) -> Option<*const c_void> {
+        self.find(key.as_concrete_TypeRef() as *const c_void)
+    }
+
+    /// # Panics
+    ///
+    /// Panics if the key is not present in the dictionary. Use `find` to get an `Option` instead
+    /// of panicking.
     #[inline]
     pub fn get(&self, key: *const c_void) -> *const c_void {
-        let value = self.find(key);
-        if value.is_none() {
-            panic!("No entry found for key {:p}", key);
+        match self.find(key) {
```

Comment:
This looks like an expanded version of `expect`.

Reply:
Indeed. I just kept the `panic!` existing before this PR. With expect I can't do the formatting as easily (will need an extra `format!`) and I did not want to remove the printing of the pointer. I don't see the point in printing the pointer but maybe there is one so I kept it. If not, the entire implementation can simply be
```rust
self.find(key).expect("No entry found for key")
```
To be honest I don't see the reason to have this `get` at all. Much more explicit to use `find` and have to `unwrap` yourself if you want panicking behavior.

I guess it would break some stuff, but I think it would be cleaner to get rid of this panicking version altogether and rename `find` to `get`. From my experience `get` is a more common name with dictionaries, plus it would match better with the underlying `CFDictionaryGetValueIfPresent`.

Reply:
Ah, so there's the `format!` macro that you can use if formatting is the problem you're facing.

Reply:
Yes. So under the conditions that `get` should remain and that it should panic with the pointer in the message, now it probably has the simplest implementation.